Nodify/Connections/BaseConnection.cs

```diff
@@ -215,21 +267,48 @@ protected override Geometry DefiningGeometry
 
         protected abstract (Point ArrowSource, Point ArrowTarget) DrawLineGeometry(StreamGeometryContext context, Point source, Point target);
 
-        protected virtual void DrawArrowGeometry(StreamGeometryContext context, Point source, Point target)
+        protected virtual void DrawArrowGeometry(StreamGeometryContext context, Point source, Point target, bool secondArrow = false)
         {
-            (Point from, Point to) = GetArrowHeadPoints(source, target);
+            (Point from, Point to) = GetArrowHeadPoints(source, target, secondArrow);
 
             context.BeginFigure(target, true, true);
             context.LineTo(from, true, true);
             context.LineTo(to, true, true);
         }
 
-        protected virtual (Point From, Point To) GetArrowHeadPoints(Point source, Point target)
+        protected virtual (Point From, Point To) GetArrowHeadPoints(Point source, Point target, bool secondArrow = false)
         {
             double headWidth = ArrowSize.Width;
             double headHeight = ArrowSize.Height;
 
-            double direction = Direction == ConnectionDirection.Forward ? 1d : -1d;
+            double direction;
+
+            switch(ArrowEnds)
+            {
```

Comment:
No need for another switch here if you send the Direction as a parameter in the first switch statement.